### PR TITLE
Switch to Gradle build system

### LINKED ISSUE #229: 1.6.3 with new support library 23.0.0 Resource NotFoundException
Getting exception when migrating to new version of the lib together with the support library update:

```
 java.lang.RuntimeException: Unable to start activity ComponentInfo{com.my.Activity}: android.view.InflateException: Binary XML file line #32: Error inflating class com.squareup.timessquare.CalendarPickerView
...
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
     Caused by: android.view.InflateException: Binary XML file line #32: Error inflating class com.squareup.timessquare.CalendarPickerView
            at android.view.LayoutInflater.createView(LayoutInflater.java:633)
            at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:743)
            at android.view.LayoutInflater.rInflate(LayoutInflater.java:806)
            at android.view.LayoutInflater.parseInclude(LayoutInflater.java:916)
...
            at android.support.v7.app.AppCompatDelegateImplV7.setContentView(AppCompatDelegateImplV7.java:257)
            at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:109)
            at com.my.Activity.onCreate(Activity.java:158)
...
     Caused by: android.content.res.Resources$NotFoundException: File res/layout/abc_action_bar_up_container.xml from color state list resource ID #0x7f040001
            at android.content.res.Resources.loadColorStateList(Resources.java:2592)
            at android.content.res.Resources.getColor(Resources.java:932)
            at com.squareup.timessquare.CalendarPickerView.<init>(CalendarPickerView.java:114)
            at java.lang.reflect.Constructor.newInstance(Native Method)
            at java.lang.reflect.Constructor.newInstance(Constructor.java:288)
            at android.view.LayoutInflater.createView(LayoutInflater.java:607)
            at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:743)
            at android.view.LayoutInflater.rInflate(LayoutInflater.java:806)
        Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #17: invalid drawable tag LinearLayout
            at android.content.res.ColorStateList.createFromXmlInner(ColorStateList.java:148)
            at android.content.res.ColorStateList.createFromXml(ColorStateList.java:131)
            at android.content.res.Resources.loadColorStateList(Resources.java:2588)
            at android.content.res.Resources.getColor(Resources.java:932)
            at com.squareup.timessquare.CalendarPickerView.<init>(CalendarPickerView.java:114)
            at java.lang.reflect.Constructor.newInstance(Native Method)
```

Tried cleaning, rebuilding with Studio and gradle, invalidating caches. If then return to 1.6.2 and 22.2.1, everything is ok.

The stack is trimmed.

### PR DESCRIPTION
**Pull request by niqo01**
Fix #229 

Some lint checks are failing.

### CODE REVIEW

**Comment by niqo01** [comment]
I am not pushing to use Gradle, this is just to demonstrate that #229 is a build issue.

**Comment by JakeWharton** [comment]
We can use Gradle, but what's worse than the "wrong" build system is having
two build systems. Delete all the Maven crap if you are truly making this
PR a proposal.

**Comment by niqo01** [comment]
Let's make this a proposal then. @JakeWharton I removed the maven related files. Build still failing on Lint.

File: .travis.yml [hunk 1]
```diff
@@ -2,8 +2,8 @@ language: android
 
 android:
   components:
-    - build-tools-20.0.0
-    - android-17
+    - build-tools-23.0.0
+    - android-21
```
Comment (JakeWharton) [inline .travis.yml]: 22

Comment (JakeWharton) [inline .travis.yml]: er, 23

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+allprojects {
+  repositories {
+    mavenCentral()
+  }
+}
+
+ext {
+  minSdkVersion = 8
+  targetSdkVersion = 21
+  compileSdkVersion = 21
```
Comment (JakeWharton) [inline build.gradle]: 23 for both of these

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  targetCompatibilityVersion = JavaVersion.VERSION_1_7
+}
+
+ext.deps = [
+    // Test dependencies
+    junit      : "junit:junit:4.10",
+    festandroid: "com.squareup:fest-android:1.0.7",
+    robolectric: "org.robolectric:robolectric:2.2",
+    intellijannotations: "com.intellij:annotations:12.0"
+
```
Comment (JakeWharton) [inline build.gradle]: nit: remove empty line

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#Fri Aug 21 12:21:00 PDT 2015
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.5-bin.zip
```
Comment (JakeWharton) [inline gradle/wrapper/gradle-wrapper.properties]: 2.6 is latest

File: sample/build.gradle [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    applicationId 'com.squareup.timessquare.sample'
+    minSdkVersion rootProject.ext.minSdkVersion
+    targetSdkVersion rootProject.ext.targetSdkVersion
+    versionCode 1
+    versionName '1.0.0'
+  }
+
+  android {
+    lintOptions {
+      ignore 'InvalidPackage'
```
Comment (JakeWharton) [inline sample/build.gradle]: is this needed?

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  sourceCompatibilityVersion = JavaVersion.VERSION_1_7
+  targetCompatibilityVersion = JavaVersion.VERSION_1_7
+}
+
+ext.deps = [
+    // Test dependencies
+    junit      : "junit:junit:4.10",
+    festandroid: "com.squareup:fest-android:1.0.7",
+    robolectric: "org.robolectric:robolectric:2.2",
+    intellijannotations: "com.intellij:annotations:12.0"
```
Comment (JakeWharton) [inline build.gradle]: we use single-quote strings for static strings and double-quote strings only when there's a template replacement (e.g., `"Name: $name"`)

File: sample/build.gradle (hunk 5 above)
Comment (niqo01) [inline sample/build.gradle]: Hum I don't see why it would. Brought it over by mistake from Timber, I ll remove.

**Comment by JakeWharton** [comment]
I think we can downgrade `MissingTranslation` to a warning. @edenman can comment on the rest of the failing lint checks.

**Comment by JakeWharton** [comment]
`.buildscript/deploy_snapshot.sh` needs updated to use Gradle. You can delete the `settings.xml` in that folder as well. The secrets will need updated for deploy to work with gradle as the format is different. I can do that post-mege.

**Pull request closed without merging**